src/libraries/Microsoft.PowerFx.Interpreter/Functions/StandardErrorHandling.cs

```diff
@@ -173,6 +173,116 @@ private static bool IsInvalidDouble(double number)
+                }
+            }
+
+            return max;
+        }
+
+        private class ExpandToSizeResult
+        {
+            public readonly string Name;
+            public readonly IEnumerable<DValue<RecordValue>> Rows;
+
+            public ExpandToSizeResult(string name, IEnumerable<DValue<RecordValue>> rows)
+            {
+                Name = name;
+                Rows = rows;
+            }
+        }
+
+        private static ExpandToSizeResult ExpandToSize(FormulaValue arg, int size)
+        {
+            string name = BuiltinFunction.ColumnName_ValueStr;
+            if (arg is TableValue tv)
+            {
+                var tvType = (TableType)tv.Type;
+                name = tvType.SingleColumnFieldName;
+
+                var count = tv.Rows.Count();
+                if (count < size)
+                {
+                    var inputRecordType = tvType.ToRecord();
+                    var inputRecordNamedValue = new NamedValue(name, new BlankValue(IRContext.NotInSource(FormulaType.Blank)));
+                    var inputRecord = new InMemoryRecordValue(IRContext.NotInSource(inputRecordType), new List<NamedValue>() { inputRecordNamedValue });
+                    var inputDValue = DValue<RecordValue>.Of(inputRecord);
+
+                    var repeated = Enumerable.Repeat(inputDValue, size - count);
+                    var rows = tv.Rows.Concat(repeated);
+                    return new ExpandToSizeResult(name, rows);
+                }
+                else
+                {
+                    return new ExpandToSizeResult(name, tv.Rows);
+                }
+            }
+            else
+            {
+                var inputRecordType = new RecordType().Add(name, arg.Type);
+                var inputRecordNamedValue = new NamedValue(name, arg);
+                var inputRecord = new InMemoryRecordValue(IRContext.NotInSource(inputRecordType), new List<NamedValue>() { inputRecordNamedValue });
+                var inputDValue = DValue<RecordValue>.Of(inputRecord);
+                var rows = Enumerable.Repeat(inputDValue, size);
+                return new ExpandToSizeResult(name, rows);
+            }
+        }
+
+        private static KeyValuePair<List<string>, List<List<DValue<RecordValue>>>> Transpose(IEnumerable<ExpandToSizeResult> results, int size)
+        {
+            var first = results.Select(result => result.Name).ToList();
+            var lists = results.Select(result => result.Rows.ToList()).ToList();
+
+            var second = new List<List<DValue<RecordValue>>>();
+
+            for (int i = 0; i < size; i++)
+            {
+                second.Add(lists.Select(list => list[i]).ToList());
+            }
+
+            return new KeyValuePair<List<string>, List<List<DValue<RecordValue>>>>(first, second);
+        }
+
+        public static Func<EvalVisitor, SymbolContext, IRContext, FormulaValue[], FormulaValue> DoubleSingleColumnTable(FunctionPtr targetFunction)
```

Comment:
What does a "DoubleSingleColumnTable" do? It's not clear by the name. Comments would help

Reply:
This has been renamed as well.

Reply:
Added comment